### PR TITLE
Semaphore Prediction for Partial Flag sets

### PR DESCRIPTION
**Pull request by beckbria**
This allows you to get a list of potential letters you can make with a set of Semaphore flags.  It also lets you query the entire Semaphore dictionary in a roundabout way by doing Predict(None).

### CODE REVIEW

File: src/Common/EncodingDataBase.ts [hunk 1]
```diff
@@ -21,6 +21,10 @@ abstract class EncodingDataBase<T extends number> {
     return result;
   }
 
+  public get entries() {
```
Comment (kfarnung) [inline src/Common/EncodingDataBase.ts]: Not really a problem, but also not necessary, a call to `lookup(SemaphoreEncoding.None)` should yield all the entries.

File: src/Semaphore/SemaphorePrediction.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import EncodingEntry from '../Common/EncodingEntry';
+import SemaphoreData from './SemaphoreData';
+import SemaphoreDirection from './SemaphoreDirection';
+import SemaphoreEncoding from './SemaphoreEncoding';
+
+class SemaphorePrediction {
```
Comment (kfarnung) [inline src/Semaphore/SemaphorePrediction.ts]: So I think this is completely unnecessary.  All classes already support prediction, simply do:

```
const ch = new SemaphoreCharacter();
const matches = ch.getPotentialMatches();
```

I'd rather continue that approach than introduce this side concept.

**Comment by kfarnung** [comment]
I would recommend doing this instead:

https://github.com/kfarnung/puzzle-lib/commit/0513dddd87096def1f8ae4ac5e1b8b3a162c5c01

File: .vscode/launch.json [hunk 3]
```diff
@@ -13,8 +13,7 @@
         "999999",
         "--colors",
         "${workspaceFolder}/test"
-      ],
-      "internalConsoleOptions": "openOnSessionStart"
```
Comment (kfarnung) [inline .vscode/launch.json]: Was this causing issues?

**Comment by beckbria** [comment]
This change is unnecessary.  I'll look into migrating the tests, but the rest I'm abandoning.

**Comment by kfarnung** [comment]
@beckbria I think migrating the tests is worth it, so don't let them get lost.

**Pull request closed without merging**